src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -48,8 +45,82 @@
         public static final SupplyCurrentLimitConfiguration hoodCurrentLimit = turretCurrentLimit;
         public static final InterpolatingTreeMap dataPoints = new InterpolatingTreeMap();
         public static final ShuffleboardTab tab = Shuffleboard.getTab("Shooter");
+        public static final NetworkTableEntry flywheelPEntry = tab.add("Flywheel P", FLYWHEEL_P).withPosition(0, 0)
+                .withSize(1, 1).getEntry();
+        public static final NetworkTableEntry flywheelIEntry = tab.add("Flywheel I", FLYWHEEL_I).withPosition(0, 1)
+                .withSize(1, 1).getEntry();
+        public static final NetworkTableEntry flywheelDEntry = tab.add("Flywheel D", FLYWHEEL_D).withPosition(0, 2)
+                .withSize(1, 1).getEntry();
+        public static final NetworkTableEntry turretPEntry = tab.add("Turret P", TURRET_P).withPosition(1, 0)
+                .withSize(1, 1).getEntry();
+        public static final NetworkTableEntry turretIEntry = tab.add("Turret I", TURRET_I).withPosition(1, 1)
+                .withSize(1, 1).getEntry();
+        public static final NetworkTableEntry turretDEntry = tab.add("Turret D", TURRET_D).withPosition(1, 2)
```

Comment:
The the PID constants will be helpful, tuning the shooter will require being able to adjust flywheel speed, hood angle and turret angle on the fly via shuffleboard.
Given the backspin on the cargo coming out of the shooter, seems like there is a decent chance we will want to put on some bias for the hood and turret angle to help keep the balls in the hoop.  Hitting the down slope inside the back of the hoop will increase the backspin and the bounce, while hitting the front or side down slopes should help deaden the bounce of the ball.

Reply:
First, I realized that I'm not actually actively updating the PID tuning for the motors (I'll do that later when I have time).
Second, I think we could incorporate the biasing when we're setting the data points for the shooting. Great idea, though.

Reply:
The advantage of the shuffleboard sliders for angle and distance are to help to be able to generate the data points.
Without being able to quickly and easily make adjustments on the fly, calibration of distance and both angles will take much, much longer (change, compile, deploy, repeat.)
So while plumbing PID updates, also do the distance and both angles.